### DISCUSSION TITLE
Long-range interaction

### DISCUSSION THREAD

**Discussion by Weizhang88** [deepmodeling/deepmd-kit#865] [Q&A]
lammps: 29Oct2020
deepmd-kit:1.3.3

Dear all,
        I want to calculate the Long-range interaction with kspace, however it stopped with the note of "**kspace style requires atom attribute q**".  Here follows the input.lammps script:
      `variable        NSTEPS          equal 2000000
variable        THERMO_FREQ     equal 200
variable        DUMP_FREQ       equal 1000
variable        TEMP            equal 300.000000
variable        PRES            equal 500
variable        TAU_T           equal 0.100000
variable        TAU_P           equal 0.500000

units           metal
boundary        p p p
atom_style      atomic

neighbor        1.0 bin

box          tilt large
read_data    conf.lmp #final.data
change_box   all triclinic

mass            1 16.000000
mass            2 1.000000

pair_style      deepmd  ../graph.001.pb ../graph.000.pb   out_freq ${THERMO_FREQ} out_file model_devi.out 
pair_coeff 
kspace_style   pppm 1.0e-5
kspace_modify  gewald 0.45     

timestep        0.0005
velocity        all create ${TEMP} 422659

thermo_style    custom step temp pe ke etotal press vol lx ly lz xy xz yz
thermo          ${THERMO_FREQ}

dump            1 all custom ${DUMP_FREQ} traj.lammpstrj id type element x y z

fix            dpgen_plm all plumed plumedfile plumed-d.dat outfile p.log
fix             1 all nvt temp ${TEMP} ${TEMP} ${TAU_T}

run             ${NSTEPS}

write_data        final.data
`
I really hope anyone can give me some help.

**Comment by mirza-galib** [comment]
In your conf.lmp file you need to provide the atomic charge. Most probably your current file has atom type and coordinates but not atom charge.
